pkg/p2p/outgoing/outgoing.go

```diff
@@ -105,7 +105,7 @@ func (a *connector) connect(ctx context.Context, c net.Conn, v proto.Version) (c
 
 	_, err = handshake.ReadFrom(c)
 	if err != nil {
-		zap.S().Debugf("Failed to read handshake: %s %s", err, a.params.Address)
+		zap.S().Debugf("[%s] Failed to read handshake: %s", a.params.Address, err)
```

Comment:
Add explicit .String() call for Address and replace error print modifier from %s to %v.